component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -632,16 +632,34 @@ private void checkStatusCode(HttpServletResponse response, AuthenticationContext
 
         boolean isRetryEnabled = SMSOTPUtils.isRetryEnabled(context);
         String loginPage = getLoginPage(context);
+        AuthenticatedUser authenticatedUser =
+                (AuthenticatedUser) context.getProperty(SMSOTPConstants.AUTHENTICATED_USER);
         String url = getURL(loginPage, queryParams);
         if (StringUtils.isNotEmpty(getScreenValue(context))) {
             url = url + SMSOTPConstants.SCREEN_VALUE + getScreenValue(context);
         }
         try {
-            if (Boolean.parseBoolean(String.valueOf(context.getProperty(SMSOTPConstants.ACCOUNT_LOCKED))) &&
-                    SMSOTPUtils.isShowAuthFailureReason(context)) {
-                response.sendRedirect(url + SMSOTPConstants.RESEND_CODE
-                        + SMSOTPUtils.isEnableResendCode(context) + SMSOTPConstants.ERROR_MESSAGE + SMSOTPConstants
-                        .ACCOUNT_LOCKED_ERROR);
+            if (SMSOTPUtils.isLocalUser(context) && SMSOTPUtils.isAccountLocked(authenticatedUser)) {
+                boolean showAuthFailureReason = SMSOTPUtils.isShowAuthFailureReason(context);
+                String retryParam;
+                if (showAuthFailureReason) {
+                    String unlockTime = getUnlockTimeInMilliSeconds(authenticatedUser);
+                    if (unlockTime == null) {
+                        if (log.isDebugEnabled()) {
```

Comment:
Shall we alter the logic as follows? WDYT?
```
if (StringUtils.isBlank(unlockTime) {
    if(log.isDebugEnabled()){
     }
} else {
   long timeToUnlock = Long.parseLong(unlockTime) - System.currentTimeMillis();
}
```

Reply:
fixed in 23e2fc0